src/preact.d.ts

```diff
@@ -277,8 +284,8 @@ declare namespace JSX {
 		charSet?:string;
 		challenge?:string;
 		checked?:boolean;
-		class?:string;
-		className?:string;
+		class?:string | Object;
+		className?:string | { [key:string]: string | boolean | number };
```

Comment:
FWIW any value in an attribute or classname object value that preact treats as Boolean will be implicitly cast, so numbers and strings are fair game there. That said, Boolean is the meaning, regardless of the type given.